src/ceph_mon.cc

```diff
@@ -786,7 +786,9 @@ int main(int argc, const char **argv)
   int rank = monmap.get_rank(g_conf()->name.get_id());
   std::string public_msgr_type = g_conf()->ms_public_type.empty() ? g_conf().get_val<std::string>("ms_type") : g_conf()->ms_public_type;
   Messenger *msgr = Messenger::create(g_ceph_context, public_msgr_type,
-				      entity_name_t::MON(rank), "mon", 0);
```

Comment:
I thought 0 was used so that one could connect to a monitor without knowing the nonce in advance?

The addresses of other daemons come from cluster maps, but monitor addresses typically come from a configuration file and don't have nonces on them.

Reply:
> I thought 0 was used so that one could connect to a monitor without knowing the nonce in advance?
> The addresses of other daemons come from cluster maps, but monitor addresses typically come from a configuration file and don't have nonces on them.

Explicit `NONCE_WILDCARD` with _nonce learning_? I'm thinking about a mechanism partially analogous to the _learning of own addr from peer_ but based on the `ServerIdentFrame`. Messenger already doesn't require there (at `handle_server_ident`) the strict equality of addrvecs precisely because of "connecting based on addresses parsed out of mon_host or something". IIUC updating `AsyncConnection::target_addr` at this stage should be fine as it is for reconnects. The wildcard value set to `0` should handle the compatibility with monmaps and older peers.
Overall a better name might be `NONCE_LEARN_LATER`.